Error in context propagation tutorial

**Describe the bug**
Unexpected behaviour when I try to use code from [Usage example for CompletionStage](https://quarkus.io/guides/context-propagation#usage-example-for-completionstage)

**Expected behavior**
Store persons to Postgre.

**Actual behavior**
Compile exception. Need to change `CompletionStage<Person> => CompletionStage<List<Person>>`


Getting error: 
[gist-link](https://gist.github.com/shvyrev/064239a6d460acd1c3a50a0207e7c499)
`RESTEASY002020: Unhandled asynchronous exception, sending back 500: java.lang.IllegalStateException: You have attempted to perform a blocking operation on a IO thread. This is not allowed, as blocking the IO thread will cause major performance issues with your application. If you want to perform blocking EntityManager operations make sure you are doing it from a worker thread.`


**To Reproduce**
Steps to reproduce the behavior:
1. use code from tutorial
2. mvn clean quarkus:dev


**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin MacBook-Pro-Sergey.local 18.7.0 Darwin Kernel Version 18.7.0: Sun Dec  1 18:59:03 PST 2019; root:xnu-4903.278.19~1/RELEASE_X86_64 x86_64
 - Output of `java -version`: OpenJDK Runtime Environment GraalVM CE 19.3.1 (build 11.0.6+9-jvmci-19.3-b07)
OpenJDK 64-Bit Server VM GraalVM CE 19.3.1 (build 11.0.6+9-jvmci-19.3-b07, mixed mode, sharing)
 - GraalVM version (if different from Java): GraalVM CE 19.3.1
 - Quarkus version or git rev: 1.2.0.Final

> @shvyrev nice catch, mind providing a PR with this fix? Here is the Asciidoc file: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/context-propagation.adoc#usage-example-for-completionstage

Sorry it's very difficult for me. I think this tutorial code never worked fine.
So many errors.

I got new one.

`10:28:28,308 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Final
10:28:29,089 INFO  [org.hib.jpa.boo.int.PersistenceXmlParser] HHH000318: Could not find any META-INF/persistence.xml file in the classpath
10:28:29,255 INFO  [org.hib.Version] HHH000412: Hibernate Core {5.4.10.Final}
10:28:29,818 ERROR [io.qua.dev.DevModeMain] Failed to start Quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type sun.jvm.hotspot.debugger.ThreadContext and qualifiers [@Default]
	- java member: org.acme.ExampleResource#threadContext
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:879)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:231)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:117)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:272)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type sun.jvm.hotspot.debugger.ThreadContext and qualifiers [@Default]
	- java member: org.acme.ExampleResource#threadContext
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:416)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:223)
	... 14 more

	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:181)
	at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:178)
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:96)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:67)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type sun.jvm.hotspot.debugger.ThreadContext and qualifiers [@Default]
	- java member: org.acme.ExampleResource#threadContext
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:879)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:231)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:117)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:272)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type sun.jvm.hotspot.debugger.ThreadContext and qualifiers [@Default]
	- java member: org.acme.ExampleResource#threadContext
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:416)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:223)
	... 14 more

	at io.quarkus.builder.Execution.run(Execution.java:108)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:128)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:114)
	... 3 more
Caused by: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type sun.jvm.hotspot.debugger.ThreadContext and qualifiers [@Default]
	- java member: org.acme.ExampleResource#threadContext
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:879)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:231)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:117)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:272)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type sun.jvm.hotspot.debugger.ThreadContext and qualifiers [@Default]
	- java member: org.acme.ExampleResource#threadContext
	- declared on CLASS bean [types=[org.acme.ExampleResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.ExampleResource]
	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:472)
	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:416)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:223)
	... 14 more

10:28:29,834 INFO  [io.qua.dev.DevModeMain] Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
10:28:29,863 ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.lang.IllegalArgumentException: workerPoolSize must be > 0
	at io.vertx.core.VertxOptions.setWorkerPoolSize(VertxOptions.java:275)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder.convertToVertxOptions(VertxCoreRecorder.java:152)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder.initializeWeb(VertxCoreRecorder.java:105)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:119)
	at io.quarkus.vertx.http.deployment.devmode.VertxHotReplacementSetup.handleFailedInitialStart(VertxHotReplacementSetup.java:30)
	at io.quarkus.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:452)
	at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:192)
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:96)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:67)
`

--- a/docs/src/main/asciidoc/context-propagation.adoc
+++ b/docs/src/main/asciidoc/context-propagation.adoc
@@ -95,16 +95,18 @@ you need manual context propagation. You can do that by injecting a `ThreadConte
 or `ManagedExecutor` that will propagate every context. For example, here we use the link:vertx[Vert.x Web Client]
 to get the list of Star Wars people, then store them in the database using
 link:hibernate-orm-panache[Hibernate ORM with Panache] (all in the same transaction) before returning
-them to the client:
+them to the client as JSON using link:rest-json[JSON-B or Jackson]:
 
 [source,java]
 --
     @Inject ThreadContext threadContext;
+    @Inject ManagedExecutor managedExecutor;
     @Inject Vertx vertx;
 
     @Transactional
     @GET
     @Path("/people")
+    @Produces(MediaType.APPLICATION_JSON)
     public CompletionStage<List<Person>> people() throws SystemException {
         // Create a REST client to the Star Wars API
         WebClient client = WebClient.create(vertx,
@@ -114,19 +116,19 @@ them to the client:
                           .setSsl(true));
         // get the list of Star Wars people, with context capture
         return threadContext.withContextCapture(client.get("/api/people/").send())
-                .thenApply(response -> {
+                .thenApplyAsync(response -> {
                     JsonObject json = response.bodyAsJsonObject();
                     List<Person> persons = new ArrayList<>(json.getInteger("count"));
                     // Store them in the DB
                     // Note that we're still in the same transaction as the outer method
                     for (Object element : json.getJsonArray("results")) {
                         Person person = new Person();
-                        person.name = ((JsonObject)element).getString("name");
+                        person.name = ((JsonObject) element).getString("name");
                         person.persist();
                         persons.add(person);
                     }
                     return persons;
-                });
+                }, managedExecutor);
     }
 --
 

